code/game/objects/items/devices/scanners/plant_analyzer.dm

```diff
@@ -0,0 +1,275 @@
+	var/list/data = list()
+	data["cycle_seconds"] = HYDROTRAY_CYCLE_DELAY / 10
+	data["trait_db"] = list()
+	for(var/trait_path in subtypesof(/datum/plant_gene/trait))
+		var/datum/plant_gene/trait/trait = new trait_path
+		var/trait_data = list(list(
+			"path" = trait.type,
+			"name" = trait.name,
+			"icon" = trait.icon,
+			"description" = trait.description
+		))
+		data["trait_db"] += trait_data
+	return data
+
+/// Called when our analyzer is used on something
+/obj/item/plant_analyzer/proc/analyze(mob/user, atom/target)
+	var/obj/item/graft/graft
+	var/obj/item/seeds/seed
+	var/obj/machinery/hydroponics/tray
+
+	playsound(src, SFX_INDUSTRIAL_SCAN, 20, TRUE, -2, TRUE, FALSE)
+
+	if (istype(target, /obj/machinery/hydroponics))
+		tray = target
+		seed = tray.myseed
+	else if (istype(target, /obj/structure/glowshroom))
+		var/obj/structure/glowshroom/shroom_plant = target
+		seed = shroom_plant.myseed
+	else if(istype(target, /obj/item/graft))
+		graft = target
+	else if (isitem(target))
+		var/obj/item/scanned_object = target
+		if(istype(scanned_object, /obj/item/seeds))
+			seed = scanned_object
+		else if (scanned_object.get_plant_seed())
+			seed = scanned_object.get_plant_seed()
+
+	if (!seed && !tray && !graft)
+		return FALSE
+
+	last_scan_data = list(
+		"tray_data" = null,
+		"seed_data" = null,
+		"graft_data" = null,
+	)
+
+	if(tray)
+		last_scan_data["tray_data"] = list(
+			"plant_health" = tray.plant_health,
+			"plant_age" = tray.age,
+			"name" = tray.name,
+			"icon" = tray.icon,
+			"icon_state" = tray.icon_state,
+			"water" = tray.waterlevel,
+			"water_max" = tray.maxwater,
+			"nutri" = tray.reagents.total_volume,
+			"nutri_max" = tray.maxnutri,
+			"yield_mod" = tray.yieldmod,
+			"weeds" = tray.weedlevel,
+			"weeds_max" = MAX_TRAY_WEEDS,
+			"pests" = tray.pestlevel,
+			"pests_max" = MAX_TRAY_PESTS,
+			"toxins" = tray.toxic,
+			"toxins_max" = MAX_TRAY_TOXINS,
+			"reagents" = list(),
+		)
+		for(var/datum/reagent/reagent in tray.reagents.reagent_list)
+			last_scan_data["tray_data"]["reagents"] += list(list(
+				"name" = reagent.name,
+				"volume" = reagent.volume
+			))
+
+	if(seed)
+		last_scan_data["seed_data"] = make_seed_data(seed)
+
+	if(graft)
+		last_scan_data["graft_data"] = list(
+			"name" = graft.parent_name,
+			"icon" = graft.icon,
+			"icon_state" = graft.icon_state,
+			"yield" = graft.yield,
+			"production" = graft.production,
+			"lifespan" = graft.lifespan,
+			"endurance" = graft.endurance,
+			"weed_rate" = graft.weed_rate,
+			"weed_chance" = graft.weed_chance,
+			"graft_gene" = graft.stored_trait.type
+		)
+
+	ui_interact(user)
+
+	return TRUE
+
+
+/obj/item/plant_analyzer/proc/make_seed_data(obj/item/seeds/seed)
+	var/list/seed_data = list(
+		"name" = seed.plantname,
+		"icon" = seed.growing_icon,
+		"icon_state" = seed.icon_harvest,
+		"product" = seed.product?.name,
+		"product_icon" = seed.product?.icon,
+		"product_icon_state" = seed.product?.icon_state,
```

Comment:
This works? Isn't product a typepath?

Reply:
It does, it considers product as `/obj/item` and it has these params